pallets/liquidity-pools-gateway-queue/src/lib.rs

```diff
@@ -202,9 +202,13 @@ pub mod pallet {
 		fn service_message_queue(max_weight: Weight) -> Weight {
 			let mut weight_used = Weight::zero();
 
-			let mut processed_entries = Vec::new();
+			let mut nonces = MessageQueue::<T>::iter_keys().collect::<Vec<_>>();
+			nonces.sort();
```

Comment:
Since we are collecting the keys here, it also made sense to me to sort them, just in case. Please let me know if there are any objections to this.

Reply:
Although this new solution is more simple, I think there is a problem here:

```rust
MessageQueue::<T>::iter_keys().collect::<Vec<_>>();
```

It can collect many keys, making the block impossible to build. 

I think we need a complex structure here that allow us to store them already organized

Reply:
> It can collect many keys, making the block impossible to build.

Can you elaborate on this please?

Reply:
How about limiting the number of keys that we collect via:
```
let mut nonces = MessageQueue::<T>::iter_keys().take(MAX_MESSAGES_PER_BLOCK).collect::<Vec<_>>();
```

Reply:
> Can you elaborate on this please?

When you collect, the iterator will make one read per item, and could be a number of items that overpass the limit for the block weight capacity.

The `take(MAX_MESSAGES_PER_BLOCK)` still does not work because could be left a message in the queue that ideally should be processed first.

Reply:
Not sure if exists some order structure available in substrate for this. If not, we should create some complex/annoying structure to organize the way the messages are stored.

But I'm not able to see a super simple way TBH. We can leave that fix for another PR to unlock this if we see it's not easy

Reply:
Maybe there's a simpler solution that involves using the latest message nonce. I'll try something on a different branch.

Reply:
Something similar to - https://github.com/centrifuge/centrifuge-chain/pull/1992